test.js

```diff
@@ -130,6 +130,68 @@ test('camelCase with pascalCase option', t => {
 	t.is(camelCase('桑德_在这里。', {pascalCase: true}), '桑德在这里。');
 });
 
+test('camelCase with preserveConsecutiveUppercase option', t => {
+	t.is(camelCase('foo-BAR', {preserveConsecutiveUppercase: true}), 'fooBAR');
+	t.is(camelCase('Foo-BAR', {preserveConsecutiveUppercase: true}), 'fooBAR');
+	t.is(camelCase('fooBAR', {preserveConsecutiveUppercase: true}), 'fooBAR');
+	t.is(camelCase('fooBaR', {preserveConsecutiveUppercase: true}), 'fooBaR');
+	t.is(camelCase('FOÈ-BAR', {preserveConsecutiveUppercase: true}), 'FOÈBAR');
+	t.is(camelCase(['foo', 'BAR'], {preserveConsecutiveUppercase: true}), 'fooBAR');
+	t.is(camelCase(['foo', '-BAR'], {preserveConsecutiveUppercase: true}), 'fooBAR');
+	t.is(camelCase(['foo', '-BAR', 'baz'], {preserveConsecutiveUppercase: true}), 'fooBARBaz');
+	t.is(camelCase(['', ''], {preserveConsecutiveUppercase: true}), '');
+	t.is(camelCase('--', {preserveConsecutiveUppercase: true}), '');
+	t.is(camelCase('', {preserveConsecutiveUppercase: true}), '');
+	t.is(camelCase('--__--_--_', {preserveConsecutiveUppercase: true}), '');
+	t.is(camelCase(['---_', '--', '', '-_- '], {preserveConsecutiveUppercase: true}), '');
+	t.is(camelCase('foo BAR?', {preserveConsecutiveUppercase: true}), 'fooBAR?');
+	t.is(camelCase('foo BAR!', {preserveConsecutiveUppercase: true}), 'fooBAR!');
+	t.is(camelCase('foo BAR$', {preserveConsecutiveUppercase: true}), 'fooBAR$');
+	t.is(camelCase('foo-BAR#', {preserveConsecutiveUppercase: true}), 'fooBAR#');
+	t.is(camelCase('XMLHttpRequest', {preserveConsecutiveUppercase: true}), 'XMLHttpRequest');
+	t.is(camelCase('AjaxXMLHttpRequest', {preserveConsecutiveUppercase: true}), 'ajaxXMLHttpRequest');
+	t.is(camelCase('Ajax-XMLHttpRequest', {preserveConsecutiveUppercase: true}), 'ajaxXMLHttpRequest');
+	t.is(camelCase([], {preserveConsecutiveUppercase: true}), '');
+	t.is(camelCase('mGridCOl6@md', {preserveConsecutiveUppercase: true}), 'mGridCOl6@md');
+	t.is(camelCase('A::a', {preserveConsecutiveUppercase: true}), 'a::a');
+	t.is(camelCase('Hello1WORLD', {preserveConsecutiveUppercase: true}), 'hello1WORLD');
+	t.is(camelCase('Hello11WORLD', {preserveConsecutiveUppercase: true}), 'hello11WORLD');
+	t.is(camelCase('РозовыйПушистыйFOOдинорогиf', {preserveConsecutiveUppercase: true}), 'розовыйПушистыйFOOдинорогиf');
+	t.is(camelCase('桑德在这里。', {preserveConsecutiveUppercase: true}), '桑德在这里。');
+	t.is(camelCase('桑德_在这里。', {preserveConsecutiveUppercase: true}), '桑德在这里。');
+});
+
+test('camelCase with both -pascalCase and -preserveConsecutiveUppercase option', t => {
```

Comment:
Why the dash?

Reply:
Oh, I just thought It would focus on the options. Never mind. That was not needed I guess.